### PR TITLE
qemu-self-test: made workers collect logs

### PR DESCRIPTION
**Pull request by JanPrusinowski**

### CODE REVIEW

File: scripts/ci/qemu-self-test.sh [hunk 1]
```diff
@@ -2,13 +2,13 @@
...
 # Define an array of commands
 commands=(
-  "robot -L TRACE -v config:qemu -v rte_ip:127.0.0.1 -v snipeit:no self-tests/setup-and-boot-menus.robot"
-  "robot -L TRACE -v config:qemu -v rte_ip:127.0.0.1 -v snipeit:no self-tests/dasharo-system-features-menus.robot"
-  "robot -L TRACE -v config:qemu -v rte_ip:127.0.0.1 -v snipeit:no self-tests/boolean-options.robot"
-  "robot -L TRACE -v config:qemu -v rte_ip:127.0.0.1 -v snipeit:no self-tests/numerical-options.robot"
-  "robot -L TRACE -v config:qemu -v rte_ip:127.0.0.1 -v snipeit:no self-tests/list-options.robot"
-  "robot -L TRACE -v config:qemu -v rte_ip:127.0.0.1 -v snipeit:no self-tests/secure-boot.robot"
-  "robot -L TRACE -v config:qemu -v rte_ip:127.0.0.1 -v snipeit:no self-tests/terminal.robot"
+  "robot -L TRACE -v config:qemu -v rte_ip:127.0.0.1 -d ./logs/$(date +%Y.%m.%d_%H.%M.%S)/setup-and-boot-menus -v snipeit:no self-tests/setup-and-boot-menus.robot"
```
Comment (macpijan) [inline scripts/ci/qemu-self-test.sh:5]: I expect this will fail as logs/ will not be present on the runner

**Comment by macpijan** [comment]
@JanPrusinowski just a reminder that we still do not collect logs

File: .github/workflows/qemu-self-test.yml [hunk 2]
```diff
@@ -43,3 +43,11 @@ jobs:
     - name: Start keywords self-tests with QEMU
       run: |
         ./scripts/ci/qemu-self-test.sh
+
+    - name: Save artifacts
+      uses: actions/upload-artifact@v4
+      with:
+        name: "qemu-logs"
+        path: |
+           ./artifacts/
```
[the remark this answers is not in the document]
Comment (JanPrusinowski) [reply .github/workflows/qemu-self-test.yml]: i have changed the folder

[the remark this answers is not in the document]
Comment (macpijan) [reply .github/workflows/qemu-self-test.yml]: Let's see if it works here: https://github.com/Dasharo/open-source-firmware-validation/actions/runs/10193509797/job/28198251927

**Pull request merged**